compiler warnings, unicode removal

hasufell opened the pull request.

hasufell committed:
Fix compiler warnings

ony commented on src/ExRender.hs:
```diff
@@ -44,11 +48,17 @@ instance Default ExCabalEnv where
+        -- unsafePerformIO is fine here, we just care about the year
+        (currentYear, _, _) = toGregorian
+                                . utctDay
+                                $ unsafePerformIO getCurrentTime
```
The whole point of `ExCabalEnv` is to provide input to renderer. Why can't either year or `exCopyright` generation be in `Main.hs` inside of `IO` monad?

Maybe we can add an option for changing whole copyright year and mentioned person.

hasufell replied:
This was a quick fix, I didn't see the point in making it more complicated and referential transparency is rather unimportant wrt the __year__.

A more complicated solution would be possible. But I am not sure I am going to implement that.

ony replied:
No problem. That was a good signal to prioritize the problem of post-editing copyright. Thus I've pushed it in #28 .

Though there is a still need to double-check generated content for something like missing license. And there is a need to make it more explicit ( #13 ).

ony commented on src/Main.hs:
```diff
@@ -16,7 +16,7 @@ import Data.Monoid
 import Data.Default
 import Data.ByteString.Lazy.Char8 (unpack)
 
-import Options.Applicative hiding (maybeReader)
```
This is specially to be compatible with optparse-applicative-0.12 since Exherbo only have `0.12.1.0` at the moment.
So this change should wait until we'll get `0.13` in `::haskell`.

Note that it produce warning only build it with `0.12`. It will not produce any warnings with `0.13`.

ony reviewed (changes requested):
Thank you for contributing to this package.

- [x] Since this package is used in Exherbo and build within Exherbo I keep dependencies to match with available pacakges. Thus I would prefer to wait until `dev-haskell/optparse-applicative` will get `0.13` version before dropping support for `0.12` version.
- [x] I'm lazy and always postpone rework of copyright header generation to be appropriate for using by other persons and not just by me. But I see no benefit in having year updated while my name and e-mail is still hardcoded there. We should allow people to specify user name preferrable with some config.
Addressed in #28 .

I agree that maybe `UnicodeSyntax` might slightly affect editing sources. But I have strong feeling that you enevitable will hit this when you start seeing many functional languages.
Can you share your expirence? What editor you are using? What problem you've hit? Is that problem still persist?

ony commented on src/ExRender.hs:
```diff
@@ -61,6 +61,7 @@ instance ExRenderPackage GenericPackageDescription where
     type ExPackageEnv GenericPackageDescription = ExCabalEnv
     exDispPkg env descr = exheres where
         nameSelf = pkgName . package $ packageDescription descr
+        ignoredPkgIds :: [PackageName]
```
Hmm.... I can see this to be already part of 7e610fe9 .

hasufell commented:
> But I have strong feeling that you enevitable will hit this when you start seeing many functional languages.
> Can you share your expirence? What editor you are using? What problem you've hit? Is that problem still persist?

I can safely say this is literally the only haskell project I've worked on with unicode syntax. No one uses it. And haskell is not Agda.

hasufell committed:
Remove UnicodeSyntax and make contributing easier

Don't require people to install weird plugins
to support this style. Keep it consistent.

hasufell commented:
> Since this package is used in Exherbo and build within Exherbo I keep dependencies to match with available pacakges. Thus I would prefer to wait until dev-haskell/optparse-applicative will get 0.13 version before dropping support for 0.12 version.

Fixed in latest patchset.

ony commented:
Thank you for re-basing and making it buildable in Exherbo.
d68af20 (compiler warnings) is merged into `master` branch.
Sorry about last commit. But I don't want to revert 16f0e38 at this moment.
Please let me know in case if you'll have issues with `UnicodeSyntax` language extension.

hasufell commented:
> Please let me know in case if you'll have issues with UnicodeSyntax language extension.

I do. It makes me not want to work on the codebase.

ony commented:
> I do. It makes me not want to work on the codebase.

Unfortunately that's an issue that cannot be addressed by simple note in `README.md`

hasufell commented:
> Unfortunately that's an issue that cannot be addressed by simple note in README.md

That's correct. Requiring people to set up a plugin just so they can work on a __single__ project and then have them mess with their editor so this stuff doesn't leak into other projects is not a reasonable thing to do.

This is definitely not common in the haskell ecosystem, neither in other functional languages, except those, that actually *do* kinda depend on unicode (e.g. Agda). In haskell, it's useless and doesn't add anything useful. Everything else is your personal preference that alienates contributors. But ofc that's your decision.
